puterbot/strategies/gisted_mixin.py

```diff
@@ -0,0 +1,49 @@
+"""
+Implements an ReplayStrategy mixin for generating LLM completions
+augmented with gisting on inputs.
+"""
+
+from loguru import logger
+import transformers as tf
+
+from puterbot.models import Recording
+from puterbot.strategies.base import BaseReplayStrategy
+
+from gisting.src import compress
+
+MAX_INPUT_SIZE = 512
+# 
+MODEL_NAME = "t5"
+
+class GistedLLMReplayStrategyMixin(BaseReplayStrategy):
+    
+    def __init__(self, recording: Recording, model_name: str = MODEL_NAME, max_input_size: int = MAX_INPUT_SIZE):
+        super.__init__(Recording)
+
+        logger.info(f"{model_name=}")
+        self.max_input_size = max_input_size
+
+
+    def get_completion(self, prompt: str):
+
+        max_input_size = self.max_input_size
+        if max_input_size and len(prompt) > max_input_size:
+            logger.warning(
+                f"Truncating from {len(prompt)=} to {max_input_size=}"
+            )
+            prompt = prompt[-max_input_size:]
```

Comment:
Isn't the purpose of gisting that we no longer have to truncate?

Reply:
I wasn't sure why we were truncating to begin with if I'm being honest, I thought this was a specification of some sort so I left it in there. I had a couple questions but, if this seems sufficient as an MVP, and if it seems correct I can create another PR after I fork the ```Gisting``` repo and then the changes inside the ```compress.py``` file will be apparent (as highlighted in the description). Do let me know how you would like me to proceed :)

Reply:
I also would like your perspective on the code inside ```compress.py``` after perhaps a cursory skim. It feels like that file is the crux of the repo but the repo author also told me to use his custom gist model subclasses and the code inside there isn't too straight forward. 

In the meantime, is there any reading or learning resource you would recommend to facilitate my understanding of code around creation, maintenance and modification of LLM's? Something hands-on or theoretical perhaps? I would love to have a stronger grasp of what I'm doing.

Reply:
Thanks @FFFiend ! Can you please share your conversation with the gisting repo author?

The best way to learn is by doing 😄